app/scripts/TrackRenderer.jsx

```diff
@@ -1958,15 +2351,23 @@ class TrackRenderer extends React.Component {
   }
 
   scrollEvent() {
+    if (!this.element) return;
     this.elementPos = this.element.getBoundingClientRect();
   }
 
-  forwardEvent(e) {
-    e.sourceUid = this.uid;
-    e.forwarded = true;
-    this.props.pubSub.publish('app.event', e);
+  /**
+   * Publishes an event to the pubSub channel, first overriding the
+   * sourceUid to be the uid of this track renderer.
+   *
+   * @param {any} event
```

Comment:
Should be the only explicit `any` in this PR. Maybe we could switch to `object`? But this function basically mutates any argument with additional fields and publishes.

Reply:
Removed, with more accurate type.